### PR TITLE
add checkpatch targets to Makefile

### PR DESCRIPTION
**Pull request by wamserma**
+ added target for calling checkpatch on working and staging area
+ added documentation for targets in Readme

thsi is a follow-up PR to https://github.com/OP-TEE/optee_test/pull/311

Signed-off-by: Markus S. Wamser <markus.wamser@mixed-mode.de>

### CODE REVIEW

**Comment by wamserma** [comment]
The Travis failure is due to checkpatch doing the Rumpelstiltskin. 
`WARNING: A patch subject line should describe the change not the tool that found it`

File: Makefile [hunk 1]
```diff
@@ -107,3 +107,12 @@ cscope:
 	${q}rm -f cscope.*
 	${q}find $(PWD) -name "*.[chSs]" | grep -v "$(PWD)/out" > cscope.files
 	${q}cscope -b -q -k
+
+.PHONY: checkpatch checkpatch-staging checkpatch-working
+checkpatch: checkpatch-staging checkpatch-working
+
+checkpatch-working:
+	@./scripts/checkpatch.sh
```
Comment (jforissier) [inline Makefile]: `${q}` instead of `@` please (and below)

Comment (wamserma) [reply Makefile]: Addressed.

**Comment by jforissier** [comment]
Merged to master with my Reviewed-by: added. Thanks!

**Pull request closed without merging**